Framework/PythonInterface/plugins/algorithms/SetSampleFromLogs.py

```diff
@@ -0,0 +1,167 @@
+
+    def PyExec(self):
+        wksp = self.getProperty("InputWorkspace").value
+        geometry = _toDict(self.getProperty("Geometry").value)
+        material = _toDict(self.getProperty("Material").value)
+        environment = _toDict(self.getProperty("Environment").value)
+        geometryContainer = _toDict(self.getProperty("ContainerGeometry").value)
+        materialContainer = _toDict(self.getProperty("ContainerMaterial").value)
+        findGeometry = self.getProperty("FindGeometry").value
+        findSample = self.getProperty("FindSample").value
+        findEnvironment = self.getProperty("FindEnvironment").value
+
+        # get a convenient handle to the logs
+        runObject = wksp.run()
+
+        # get information for the material from the logs
+        if findSample:
+            if (not _hasValue(material, 'ChemicalFormula')) and _hasValue(runObject, "SampleFormula"):
+                self.log().information("Looking for 'SampleFormula' in logs")
+                material['ChemicalFormula'] = _getLogValue(runObject, 'SampleFormula').strip()
+            if (not _hasValue(material, "SampleMassDensity")) and _hasValue(runObject, "SampleDensity"):
+                self.log().information("Looking for 'SampleDensity', 'SampleMass', and 'SampleMassDensity' in logs")
+                value = _getLogValue(runObject, 'SampleDensity')
+                if value == 1.0 or value == 0.0:
+                    material['Mass'] = _getLogValue(runObject, 'SampleMass')
+                elif _hasValue(runObject, "SampleMass"):
+                    material['SampleMassDensity'] = value
+        self.log().information('MATERIAL: ' + str(material))
+
+        # get height of sample from the logs
+        # this assumes the shape has a "Height" property
+        if findGeometry:
+            instrEnum = ConfigService.getInstrument(wksp.getInstrument().getFullName())
+            if not _hasValue(geometry, "Height"):
+                heightInContainerNames = ['HeightInContainer']
+                heightInContainerUnitsNames = ['HeightInContainerUnits']
+
+                # determine "special" logs for SNS instruments
+                if instrEnum.facility().name() == 'SNS':
+                    beamline = ''
+                    # TODO this would benefit from the beamline exposed through python
+                    if instrEnum.name() == 'NOMAD':
+                        beamline = 'BL1B'
+                    elif instrEnum.name() == 'POWGEN':
+                        beamline = 'BL11A'
+                    else:
+                        warningMsg = 'Do not know how to create lognames for "{}"'.format(instrEnum.name())
+                        self.log().warn(warningMsg)
+                    if beamline:
+                        heightInContainerUnitsNames.append('{}:CS:ITEMS:HeightInContainerUnits'.format(beamline))
+                        heightInContainerNames.append('{}:CS:ITEMS:HeightInContainer'.format(beamline))
```

Comment:
I am not sure if NOMAD and POWGEN share the exact same log names.
If so, we might to make it clear here with a comment.